homeassistant/components/vilfo/sensor.py

```diff
@@ -0,0 +1,105 @@
+"""Support for Vilfo Router sensors."""
+from homeassistant.helpers.entity import Entity
+
+from .const import (
+    ATTR_ICON,
+    ATTR_LABEL,
+    ATTR_ROUTER_MANUFACTURER,
+    ATTR_ROUTER_MODEL,
+    ATTR_ROUTER_NAME,
+    ATTR_UNIT,
+    DOMAIN,
+    SENSOR_TYPES,
+)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add Vilfo Router entities from a config_entry."""
+    vilfo = hass.data[DOMAIN][config_entry.entry_id]
+
+    sensors = []
+
+    name = ATTR_ROUTER_NAME
+    device_info = {
+        "identifiers": {(DOMAIN, vilfo.host, vilfo.mac_address)},
+        "name": name,
+        "manufacturer": ATTR_ROUTER_MANUFACTURER,
+        "model": ATTR_ROUTER_MODEL,
```

Comment:
A name is different from the model.

And `ATTR_ROUTER_NAME` == `ATTR_ROUTER_MODEL` here.

The model should be the "hardware version", remove it here.

Reply:
Ah, ok! I think I understand the difference.

In this case, I think there is only one hardware version available of the router (to this date).  Would it be better to change the name into something else or more generic? Like within this integration, the router is basically that -- "Router".

I'm not sure what's the best practice when it comes to device tracker and connected devices, but Vilfo also has the concept of "VPN groups" which can have connection statuses etc. Would these "VPN groups" be suitable to add as additional entities later on (as children to the router).

Just trying to wrap my head around the concepts here 😄

Reply:
@Quentame I gave this some more thought (regarding the `name` being the same as the `model`).

Since the `name` can be overridden/customized by the user, it think it makes sense to have the `model` indicating that it is indeed a `Vilfo Router` and nothing else. Does that sound reasonable? 🙂 

Do you have any input on this, @MartinHjelmare ?

Reply:
The items here should represent the device and not the entity. If it makes sense for the device, it's good. I don't know the device so I can't give specific advice.